alacritty/src/display/hint.rs

```diff
@@ -378,7 +381,31 @@ impl<'a, T> HintPostProcessor<'a, T> {
             }
         }
 
-        start..=iter.point()
+        if start > iter.point() {
+            None
+        } else {
+            Some(start..=iter.point())
+        }
+    }
+
+    /// Loops over processed matches until a match is found or the
+    /// range is exhausted.
+    fn iterate_to_next_processed_match(&mut self, mut regex_match: Match) {
+        self.next_match = None;
+
+        while regex_match.start() <= regex_match.end() {
+            self.next_match = self.hint_post_processing(&regex_match);
+
+            regex_match = match self.next_match {
+                Some(_) => break,
+                None => regex_match.start().add(self.term, Boundary::Grid, 1)..=*regex_match.end(),
+            }
```

Comment:
I'm not sure if this contradicts something I've said previously, but I don't think overlapping hint matches should be post-processed multiple times. The `regex_search_right` method is already responsible for handling potential overlaps.

So really if `hint_post_processing` doesn't return a match, the next match must be determined by regex search not by post processing.